app/Plugins/Stripe/Controllers/SettingsController.php

```diff
@@ -182,28 +180,27 @@ public function postPaymentWithStripe(Request $request)
                     $message = $view['message'];
                 }
 
-                return redirect()->back()->with($status, $message);
+                return redirect('checkout')->with($status, $message);
             } catch (\Exception $ex) {
-                dd($ex);
                 throw new \Exception($ex->getMessage(), $ex->getCode(), $ex->getPrevious());
             }
 
-                /**
-                * Write Here Your Database insert logic.
-                */
                 \Session::put('success','Money add successfully in wallet');
                 return redirect()->route('stripform');
             } else {
-                \Session::put('error','Money not add in wallet!!');
-                return redirect()->route('stripform');
+                \Session::put('fails','Your Payment was declined. Do you want to try making payment with the other gateway?');
+                return redirect()->route('checkout');
             }
         } catch (Exception $e) {
+            \Session::put('amount',$request['amount']);
             \Session::put('error',$e->getMessage());
             return redirect()->route('stripform');
         } catch(\Cartalyst\Stripe\Exception\CardErrorException $e) {
```

Comment:
`CardErrorException` must be extending to `Exception`. In that case, it will never enter this block. Same goes for `MissingParameterException`

Reply:
Fixed